src/core/unit.h

```diff
@@ -759,7 +759,7 @@ static inline const UnitVTable* UNIT_VTABLE(const Unit *u) {
 
 /* For casting a unit into the various unit types */
 #define DEFINE_CAST(UPPERCASE, MixedCase)                               \
-        static inline MixedCase* UPPERCASE(Unit *u) {                   \
+        static inline MixedCase* UPPERCASE(const Unit *u) {             \
```

Comment:
honestly, I'd instead have changed the prototype of service_can_live_mount(). I find it very unconvincing that such a helper shouldn't be able to make changes to a Unit object, for example to cache stuff.